### DISCUSSION TITLE
Proper way to test actor private field value changed with pytest

### DISCUSSION THREAD

**Discussion by SomeAkk** [jodal/pykka#108] [Q&A] [answered]
Hello!
I got actor with method which change private field of that actor, it looks like that:

```
class BufferActorClass(pykka.ThreadingActor):

    def __init__(self):
        super().__init__()
        self._buffer = []

    def buffer_flush(self):
        for message in self._buffer:
            self.actor_ref.tell(message)
        self._buffer = []
```
Now i writing unit test for that function: test check if `buffer_flush` clear `_buffer` or not. Look at code below:
```
def mocked_consumer_actor():
    pass

def mocked_buffer_actor_with(buffer):
    proxy = BufferActor.start(parent_actor=mocked_consumer_actor).proxy()
    proxy._actor._buffer = buffer
    return proxy

class TestBufferActorClassMethods:

    @pytest.mark.parametrize(
        argnames="buffer",
        argvalues=(
            ["message1", "message2", "message3"],
            ["message1"],
            []
        ),
    )
    def test_buffer_flush_must_clear_buffer(self, buffer):
        actor = mocked_buffer_actor_with(buffer)
        actor._actor._buffer = buffer
        actor.buffer_flush().get()
        assert actor._actor._buffer is []
```
After `actor.buffer_flush().get()` i am waiting that `_buffer` will be `[]`, but this not going such, instead i got `["message1", "message2", "message3"]` or another pytest parameters. What is wrong in my test? How to propper check field states of actors?

**Updated**
Also i am found that this problem coased by `self.actor_ref.tell(message)` - if delete this line all is ok and test is passed.

**Comment by SomeAkk** [comment, marked as answer]
Found problem: it was in my code, any help unneeded)
Problem was that after self.actor_ref.tell(message) in on_receive() i added messages in `._buffer` again. 
So in test i saw buffer with messages `["message1", "message2", "message3"]` again)

**Reply by jodal** [reply]
Thanks for updating you question with the solution!
